org.eclipse.xtext/src/org/eclipse/xtext/resource/impl/LiveShadowedResourceDescriptions.java

```diff
@@ -43,15 +44,18 @@
 public class LiveShadowedResourceDescriptions extends ResourceSetBasedResourceDescriptions {
 	
 	@Inject
-	private ResourceSetBasedResourceDescriptions localDescriptions;
+	protected ResourceSetBasedResourceDescriptions localDescriptions;
 
 	@Inject
-	private IResourceDescriptions globalDescriptions;
+	private Provider<IResourceDescriptions> globalDescriptionsProvider;
 
+	protected IResourceDescriptions globalDescriptions;
```

Comment:
same here

Reply:
Because we set it in the new subclass.

Reply:
and whats the setter for?

Reply:
Would it make sense to have a setter like method that accepts both values as arguments rather than breaking the encapsulation?